src/sync/sync_manager.hpp

```diff
@@ -66,7 +66,8 @@ friend class SyncSession;
     // Configure the metadata and file management subsystems. This MUST be called upon startup.
     void configure_file_system(const std::string& base_file_path,
                                MetadataMode metadata_mode=MetadataMode::Encryption,
-                               util::Optional<std::vector<char>> custom_encryption_key=none);
+                               util::Optional<std::vector<char>> custom_encryption_key=none,
+                               bool reset_metadata_on_error=false);
```

Comment:
Do any bindings want that behavior? The method used for resetting the metadata (`SyncFileManager::remove_metadata_realm`) is internal to the object store so it's not clear to me how a binding could usefully reset the sync metadata when handling errors itself.